proposals/0013-error-function.md

```diff
@@ -0,0 +1,57 @@
+# Add 'error' function
+
+|||
+|---|---
+| **JEP**    |  20
+| **Author** | Sebastien Rosset
+| **Created**| 22-March-2023
+| **[SemVer](https://semver.org/spec/v2.0.0.html#summary)** | MINOR
+
+## Abstract
+
+This JEP proposes a new function `error()` that raises a runtime error.
+This could be used to detect and raise an error when input JSON documents have unexpected data.
+
+
+## Motivation
+
+AFAIK, there is no dedicated function to raise a runtime error within a JMESpath expression.
```

Comment:
The motivation section should go into more detail as to `why` JMESPath expressions need the ability to raise a runtime error.  The existing errors are intended for cases where the user has made a mistake in their expression.  JMESPath is primarily a language for filtering/extracting/reshaping JSON data, and not really about schema validation.  This section should make a convincing case for that.

Reply:
I have added a section with concrete use cases and improved the motivation section.

Reply:
@jamesls , does this address your concerns?